matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/EventRelationsAggregationTask.kt

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package im.vector.matrix.android.internal.session.room
+
+import arrow.core.Try
+import com.zhuinden.monarchy.Monarchy
+import im.vector.matrix.android.api.session.events.model.*
+import im.vector.matrix.android.api.session.room.model.message.MessageContent
+import im.vector.matrix.android.api.session.room.model.relation.ReactionContent
+import im.vector.matrix.android.api.session.room.send.SendState
+import im.vector.matrix.android.internal.database.mapper.ContentMapper
+import im.vector.matrix.android.internal.database.mapper.EventMapper
+import im.vector.matrix.android.internal.database.model.*
+import im.vector.matrix.android.internal.database.query.create
+import im.vector.matrix.android.internal.database.query.where
+import im.vector.matrix.android.internal.task.Task
+import im.vector.matrix.android.internal.util.tryTransactionAsync
+import io.realm.Realm
+import timber.log.Timber
+
+internal interface EventRelationsAggregationTask : Task<EventRelationsAggregationTask.Params, Unit> {
+
+    data class Params(
+            val events: List<Pair<Event, SendState>>,
+            val userId: String
+    )
+}
+
+/**
+ * Called by EventRelationAggregationUpdater, when new events that can affect relations are inserted in base.
+ */
+internal class DefaultEventRelationsAggregationTask(private val monarchy: Monarchy) : EventRelationsAggregationTask {
+
+    override fun execute(params: EventRelationsAggregationTask.Params): Try<Unit> {
+        return monarchy.tryTransactionAsync { realm ->
+            update(realm, params.events, params.userId)
+        }
+    }
+
+    fun update(realm: Realm, events: List<Pair<Event, SendState>>?, userId: String) {
```

Comment:
private function?

also `events` should not be nullable

Reply:
done